hyperspy/axes.py

```diff
@@ -971,12 +965,14 @@ def get_axis_dictionary(self):
 
     def convert_to_non_uniform_axis(self):
         d = super().get_axis_dictionary()
+        axes_manager = self.axes_manager
         d["_type"] = 'DataAxis'
         self.__class__ = DataAxis
         self.__init__(**d, axis=self.axis)
         del self._expression
         del self._function
         self.remove_trait('x')
+        self.axes_manager = axes_manager
```

Comment:
I added this, because otherwise axes were loosing the `is_binned` attribute - I guess I should add a test that checks that `is_binned`, `navigate` and the like remain the same.

Reply:
done

Reply:
Right, I mixed up the attributes.